File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C#/Problem2.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace AbuHadHoud
{
    internal class Program
    {
        static string ReadName()
        {
            string Name;
            Console.Write("Please enter your Name : ");
            Name = Console.ReadLine();
            return Name;
        }

        static void PrintName(string Name)
        {
            Console.WriteLine($"\nYour Name Is : {Name}");
        }
        static void Main(string[] args)
        {
            PrintName(ReadName());
        }
    }
}

File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/Abu_HadHoud_Solutions_Using_C++/Problem2.cpp
#include <iostream>
#include <string>

using namespace std;

string ReadName() {
	string Name;

	cout << "please enter your name : " << endl;
	getline(cin, Name);
	return Name;
}

void PrintName(string Name) {
	cout << "\nYour Name is : " << Name << endl;
}

int main() {
	PrintName(ReadName());
}